src/xbot/common/math/PID.java

```diff
@@ -23,6 +28,37 @@ public void reset()
     {
         m_prevError = 0;
         m_totalError = 0;
+        errorIsSmall = false;
+        derivativeIsSmall = false;
+    }
+    
+    /**
+     * Set how close the error can be before it is considered
+     * "on-target."
+     *            
+     * This is in the same units as your current and goal values.
+     */
+    public void setErrorTolerance(double errorTolerance) {
```

Comment:
Sounds reasonable. I'll split it into enable and values and try it on the robot for a while, see how the flow feels.

(In this particular case I think it makes sense to consider SD a major consumer, since almost all of our PID values are tuned/modified using it.)

I'm not sure what the proper response is to negative values being set programmatically or via the SD. Do we want to intercept and invert them? Silently ignore, falling into the "never finishes" condition?